core/src/main/java/org/hisp/dhis/android/core/datavalue/DataValueCollectionRepository.java

```diff
@@ -38,25 +38,27 @@
 final class DataValueCollectionRepository extends ReadOnlyCollectionRepositoryImpl<DataValue>
         implements ReadWriteCollectionRepository<DataValue> {
 
+    private final DataValueStore dataValueStore;
```

Comment:
@AbelCS @luisyng store is private in ReadOnlyCollectionRepositoryImpl. What about making it protected in ReadOnlyCollectionRepositoryImpl so we don't duplicate variables?

Reply:
Yes, that makes a lot of sense.

Reply:
Well, in this case it does not work because it uses the method "exists" in DataValueStore, which is not generic